src/Misc/Matrix.jl

```diff
@@ -882,6 +882,17 @@ end
 
 diagonal_matrix(R::Ring, x::Vector{<:RingElement}) = diagonal_matrix(R.(x))
 
+function diagonal_matrix(R::Ring, x::Vector{<:MatElem})
+  if any(base_ring(i)!==R for i in x)
+    x = [change_base_ring(R,i) for i in x]
+  end
+  if length(x)>0
+    return diagonal_matrix(x)
+  else
+    return zero_matrix(R,0,0)
+  end
+end
+
```

Comment:
This should be added in AbstractAlgebra or Nemo (for the time until the remains of https://github.com/Nemocas/Nemo.jl/pull/1501 have been sorted out there)